### PR TITLE
Bug 1882197: Add test for golden image upload

### PR DESCRIPTION
**Pull request by gouyang**
note: create vm from the golden OS is not included in this test.

### CODE REVIEW

File: frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+    await browser.wait(
+      until.textToBePresentInElement(
+        $('.co-m-pane__heading'),
+        'Upload Data to Persistent Volume Claim',
+      ),
+    );
+  });
+
+  afterAll(async () => {
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts:47]: can we group afterAll and beforeAll together for easier visibility? :)

File: frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    execSync(`kubectl delete -n ${testName} dv ${pvc.name}`);
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts:48]: what about the upload token, if something goes wrong? @glekner is there something else we should take care about?

File: frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  });
+
+  it(
+    'ID(CNV-4718) Upload data to CDI',
+    async () => {
+      await fillInput(inputPVCName, pvc.name);
+      await inputPVCSize.sendKeys(pvc.size);
+
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts:56]: so we use the default SC? Don't we wanna use the SC from env variable as discussed in last PR?

File: frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+describe('KubeVirt CDI Upload', () => {
+  const pvc = { name: `upload-pvc-${testName}`, size: 1 };
+  const srcImage =
+    'http://cnv-qe-server.rhevdev.lab.eng.rdu2.redhat.com/files/cnv-tests/cirros-images/cirros-0.4.0-x86_64-disk.qcow2';
+  const desImage = '/tmp/cirros.qcow2';
+  const invalidImage = '/tmp/cirros.txt';
+
+  beforeAll(async () => {
+    execSync(`curl ${srcImage} -o ${desImage}`);
+    execSync(`cp ${desImage} ${invalidImage}`);
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts:30]: symlink would be more lightweight here.

File: frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+      });
+      await cdiUploadView.uploadInput.sendKeys(desImage);
+      await click(saveChangesBtn);
+      await browser.wait(until.textToBePresentInElement(cdiUploadView.uploadProgress, '100%'));
+    },
+    CDI_UPLOAD_TIMEOUT_SECS,
+  );
+
+  it(
+    'ID(CNV-4891) It shows error messsages when image format is not supported',
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts:70]: these functions are very similar. Can we do something about it?

File: frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+      await inputPVCSize.sendKeys(pvc.size);
+
+      // firefox needs the input to be shown
+      await browser.executeAsyncScript((callback) => {
+        (document.querySelector('input[type="file"]') as HTMLElement).style.display = 'inline';
+        callback();
+      });
+      await cdiUploadView.uploadInput.sendKeys(desImage);
+      await click(saveChangesBtn);
+      await browser.wait(until.textToBePresentInElement(cdiUploadView.uploadProgress, '100%'));
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts:64]: can we check the final PVC exists?

File: frontend/packages/kubevirt-plugin/integration-tests/views/cdiUploadView.ts [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import { element, by, $, $$ } from 'protractor';
+
+export const uploadCdiFormButton = element(by.partialButtonText('Data upload'));
+export const uploadInput = $('input[type="file"]');
+export const unitDropdown = $('button[data-test-id="dropdown-button"]');
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/integration-tests/views/cdiUploadView.ts:5]: there are some variables which are not used. Can you please remove them?

**Review by suomiy — CHANGES_REQUESTED** [review]

File: frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts (hunk 2 above)
Comment (gouyang) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts:48]: it doesn't need to handle the upload token, the framework handle it automatically.

File: frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts (hunk 3 above)
Comment (gouyang) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts:56]: ok, use SC from env.

File: frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts (hunk 4 above)
Comment (gouyang) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts:30]: ok

File: frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts (hunk 6 above)
Comment (gouyang) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts:47]: Add a step to verify the PVC is in Bound status.

**Comment by gouyang** [comment]
@suomiy I rewrite the tests, could you please review again?

File: frontend/packages/kubevirt-plugin/integration-tests/tests/models/pvcUploadForm.ts [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+  storageclassDropdown,
+  selectAccessMode,
+  sizeUnitsDropdown,
+} from '@console/ceph-storage-plugin/integration-tests/views/pvc.view';
+import { click, fillInput } from '@console/shared/src/test-utils/utils';
+import { selectOptionByText } from '../utils/utils';
+import * as cdiUploadView from '../../views/cdiUploadView';
+import { PVCData } from '../types/pvc';
+
+export class UploadForm {
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/models/pvcUploadForm.ts:22]: nice refactoring :)

File: frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts (hunk 2 above)
Comment (suomiy) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts]: I am wondering if such objects can stay alive if the execution fails?

File: frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+  const invalidImage = '/tmp/cirros.txt';
+  const pvcName = `upload-pvc-${testName}`;
+
+  beforeAll(async () => {
+    execSync(`curl ${srcImage} -o ${desImage}`);
+    execSync(`ln -s -f ${desImage} ${invalidImage}`);
+  });
+
+  afterAll(async () => {
+    execSync(`kubectl delete -n ${testName} dv ${pvcName}`);
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts:31]: shouldn't we also delete the downloaded image?

**Review by suomiy — CHANGES_REQUESTED** [review]

File: frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts (hunk 2 above)
Comment (gouyang) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts]: It stays alive if the execution fails until the ns is removed.

File: frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts (hunk 9 above)
Comment (gouyang) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts:32]: Updated, delete the downloaded image also.

File: frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts (hunk 2 above)
Comment (suomiy) [reply frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts]: ok

**Pull request merged**